jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -215,6 +225,43 @@ public Entry acquire()
         return null;
     }
 
+    /**
+     * Utility method to acquire an entry from the pool,
+     * reserving and creating a new entry if necessary.
+     *
+     * @param creator a function to create the pooled value for a reserved entry.
+     * @return an entry from the pool or null if none is available.
+     */
+    public Entry acquire(Function<Pool<T>.Entry, T> creator)
+    {
+        Entry entry = acquire();
+        if (entry != null)
+            return entry;
+
+        entry = reserve(getMaxEntries());
```

Comment:
Minor nit: you probably should call `reserve(-1)` to make it more obvious that you do not want to be limited by the pending/disabled entries count.